dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -146,13 +155,25 @@ def _determine_pf_parts(fs, paths, gather_statistics, **kwargs):
                 pf.cats = paths_to_cats(fns, scheme)
                 parts = paths.copy()
     elif fs.isdir(paths[0]):
-        # This is a directory, check for _metadata, then _common_metadata
-        paths = fs.glob(paths[0] + fs.sep + "*")
-        paths, base, fns = _sort_and_analyze_paths(paths, fs)
-        if "_metadata" in fns:
+        # This is a directory.
+        # Check if _metadata and/or _common_metadata files exists
+        base = paths[0]
+        _metadata_exists = fs.isfile(fs.sep.join([base, "_metadata"]))
```

Comment:
If `ignore_metadata_file`, then we know we will have to list the whole directory anyway, so we can avoid the calls to `isfile` and even `isdir`, above.

Reply:
Good point - I just pushed a change to avoid the `isfile` calls.  However, it seems like we may need to move around the logic quite a bit to avoid the `isdir` call.  In fact, I'm nor really sure of an elegant way to avoid it.